### PR TITLE
Add parsing file for graph creation

### LINKED ISSUE #4: Add parsing file for graph creation

### PR DESCRIPTION
**Pull request by GabinDDL**
Add functions to parse the standard input when executing the project, and add a script to compare output and expected output files when executing input files.

Close #4

### CODE REVIEW

File: bin/main.ml [hunk 1]
```diff
@@ -1 +1,16 @@
-let () = print_endline "Hello World!"
+open Prjt_mi_recolor.Graph
+open Prjt_mi_recolor.Parser
+
+let run_solver_on_file file =
```
Comment (ThomArr) [inline bin/main.ml:4]: Maybe give 2 different files, 1 for input and the other for output?

File: bin/main.ml [hunk 2]
```diff
@@ -1 +1,16 @@
...
+  match parse_file_for_solver file with
+  | Error e -> raise e
+  | Ok ((l, w), a) ->
+      let grid = init_toroidal_grid 0 l w a in
+      pp_graph Format.std_formatter grid
+(* TODO: To change with the result of SAT solver. *)
+
+let () =
+  let l = Array.length Sys.argv in
+  if l > 3 || l <= 1 then failwith "Not a good number of arg"
```
Comment (ThomArr) [inline bin/main.ml:15]: Maybe give 2 different files, 1 for input and the other for output?

File: lib/graph.ml [hunk 3]
```diff
@@ -75,17 +75,14 @@ let get_neighbors_toroidal_grid t w l x y a =
     (n4, t, a.(snd n4).(fst n4));
   ]
 
-let init_toroidal_grid t w l a =
+let init_toroidal_grid t l w a =
```
Comment (ThomArr) [inline lib/graph.ml:78]: I preferred the old version row and column to the new version column and row. But as you want.

File: README.md [hunk 4]
```diff
@@ -75,9 +75,21 @@ instead.
...
 ```
 
+The file must be in the following format:
+```
+l w
+n11 n12 n13 ... n1l
+n21 n22 ... ... ...
+n31 ... ... ... ...
+... ... ... ... ...
+nw1 nw2 nw3 ... nwl
```
Comment (ThomArr) [inline README.md:88]: Perhaps we are not obliged to specify the coordinates, as is implicit?

File: lib/Parser/parser.ml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+exception Wrong_number_of_args
+
+let parse_argument_line_to_array (l : string) : int array =
+  let str_list = String.split_on_char ' ' l in
```
Comment (ThomArr) [inline lib/Parser/parser.ml:4]: Maybe add a check of the size of the table with the dimension?

File: lib/Parser/parser.ml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  let int_list = List.map (fun x -> int_of_string x) str_list in
```
Comment (ThomArr) [inline lib/Parser/parser.ml:5]: I come back here on how to encode variables in the text file. If we have "1112" to say color 11 row 1 column 2, how do we differentiate it from color 1 row 11 column 2 (or other variants). I really think the coordinates are too much in the model (column and row numbers are implicit, so not mandatory to specify).

File: README.md (hunk 4 above)
Comment (ThomArr) [reply README.md:88]: https://github.com/GabinDDL/Math-Info-L3-Project/pull/15/files#r1524331279

File: lib/graph.ml (hunk 3 above)
Comment (GabinDDL) [reply lib/graph.ml:78]: you're right I'll change that again

File: README.md (hunk 4 above)
Comment (GabinDDL) [reply README.md:88]: > Perhaps we are not obliged to specify the coordinates, as is implicit?

I'm having a bit of trouble understanding, basically I specify for the 1st line the number of rows and columns, and then how the table is and the n represents the color and only the color of the coordinates where it is, for example n11 if the box of coordinates 1 1 must be color 2, will be equal to 2

**Comment by GabinDDL** [comment]
@ThomArr You can review, I've corrected what you said me

File: bin/main.ml [hunk 7]
```diff
@@ -1 +1,24 @@
...
+      let grid_final = init_toroidal_grid t w l a_final in
+      match file_output with
+      | None -> Format.printf "%a\n\n%a" pp_graph grid_init pp_graph grid_final
+      | Some output ->
+          let ic = open_out output in
+          Format.fprintf
+            (Format.formatter_of_out_channel ic)
+            "%a\n\n%a" pp_graph grid_init pp_graph grid_final;
+          close_out ic)
+(* TODO: To change with the result of SAT solver. *)
```
Comment (ThomArr) [inline bin/main.ml:18]: Maybe do that now?

File: lib/dune [hunk 8]
```diff
@@ -1,3 +1,5 @@
+(include_subdirs qualified)
```
Comment (ThomArr) [inline lib/dune:1]: Just, what’s that for?

**Pull request merged**